modules/library/jdbc/src/main/java/org/geotools/jdbc/JDBCDataStore.java

```diff
@@ -4147,17 +4242,43 @@ protected void encodeGroupByStatement(
 
     protected void encodeFunction(
             String function, Expression expression, StringBuffer sql, FilterToSQL filterToSQL) {
-        sql.append(function).append("(");
-        if (expression == null) {
+        encodeFunction(function, Arrays.asList(expression), sql, filterToSQL);
+    }
+
+    protected void encodeFunction(
+            String function,
+            List<Expression> expressions,
+            StringBuffer sql,
+            FilterToSQL filterToSQL) {
+        if (expressions == null || expressions.isEmpty()) {
+            sql.append(function);
+            sql.append("(");
             sql.append("*");
+            sql.append(")");
         } else {
             try {
-                sql.append(filterToSQL.encodeToString(expression));
+                int size = expressions.size();
+                boolean encodeOnce = isEncodeOnceFunction(function);
```

Comment:
So now it seems to be encoding something like:

``distinct(att1), distinct(att2), ...``

Wondering, is there a test actually covering this code path with multiple attributes?

Reply:
there is a boolean as a result of a method called isEncodeOnceFunction(function) checking if the function should be encoded once or multiple time. The result will be DISTINCT (att1), (att2), ....